components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/mappings/SubscriptionMappingUtil.java

```diff
@@ -90,6 +87,60 @@ public static SubscriptionListDTO fromSubscriptionListToDTO(List<SubscribedAPI>
         return subscriptionListDTO;
     }
 
+    /**
+     * Converts a List object of SubscribedAPIs into a DTO
+     *
+     * @param subscriptions a list of SubscribedAPI objects
+     * @param query query to filter subscriptions
+     * @return SubscriptionListDTO object containing SubscriptionDTOs
+     */
+    public static SubscriptionListDTO fromSubscriptionListToDTO(List<SubscribedAPI> subscriptions, String query)
+            throws APIManagementException {
+        SubscriptionListDTO subscriptionListDTO = new SubscriptionListDTO();
+        List<SubscriptionDTO> subscriptionDTOs = subscriptionListDTO.getList();
+        if (subscriptionDTOs == null) {
+            subscriptionDTOs = new ArrayList<>();
+            subscriptionListDTO.setList(subscriptionDTOs);
+        }
+
```

Comment:
Remove unwanted line

Reply:
This line was added for clarity to separate initializations from the logic